src/averageAges.js

```diff
@@ -55,10 +69,25 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const filteredPeople = onlyWithSon
+    ? people.filter(person => person.sex === 'm'
+      && person.mother !== null)
+    : people.filter(person => person.mother !== null);
```

Comment:
```suggestion
  const filteredPeople =  people.filter(person => onlyWithSon ? person.sex === 'm'
      && person.mother !== null : person.mother !== null) )
```
better to make ternary inside filter